common/src/main/java/org/jboss/pnc/common/concurrent/Sequence.java

```diff
@@ -20,13 +20,47 @@
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
+/**
+ * A class for generating 64-bit unique IDs at high scale. The IDs generated by this service are roughly time sortable.
+ *
+ * The ID generation follows Twitter snowflake pattern and the IDs are made up of the following components:
+ *
+ * Epoch timestamp in millisecond precision - 41 bits (gives us 69 years with a custom epoch) Configured machine id - 10
+ * bits (gives us up to 1024 machines) Sequence number - 12 bits (A local counter per machine that rolls over every
+ * 4096)
+ */
 public class Sequence {
 
-    private static final Map<Integer, SequenceGenerator> instance = new ConcurrentHashMap<>();
-    private static final int INSTANCE_KEY = 0;
+    private static final Map<Integer, Sequence> instance = new ConcurrentHashMap<>();
+
+    /**
+     * If nodeId is used, it can be set only once and have to be set before the first nextId() call. If nodeId is not
+     * explicitly defined it's calculated based on machines mac address.
+     *
+     * @param nodeId an int value between (inclusive) 0 and 1023.
+     */
+    public static void setNodeId(int nodeId) {
+        getInstance(nodeId);
+    }
 
     public static Long nextId() {
-        SequenceGenerator sequenceGenerator = instance.computeIfAbsent(INSTANCE_KEY, k -> new SequenceGenerator());
-        return sequenceGenerator.nextId();
+        return getInstance(null).sequenceGenerator.nextId();
+    }
+
+    private static Sequence getInstance(Integer nodeId) {
+        return instance.computeIfAbsent(0, k -> new Sequence(nodeId));
```

Comment:
Make the `0` constant.

Reply:
What the benefit of using the constant, it's used only here and it can be anything ?

Reply:
1) readability - giving it a name may be less confusing ("Why is there zero? Why not one?")
2) negligible speed - making it Integer constant removes the autoboxing of int.

gcode with autoboxing:
```
       3: iconst_0
       4: invokestatic  #3                  // Method java/lang/Integer.valueOf:(I)Ljava/lang/Integer;
```
gcode with constant:
```
       3: getstatic     #3                  // Field KEY:Ljava/lang/Integer;
```